src/Microsoft.AspNetCore.Server.Kestrel/Internal/Http/FrameOfT.cs

```diff
@@ -92,6 +92,13 @@ public class Frame<TContext> : Frame
                         try
                         {
                             await _application.ProcessRequestAsync(context).ConfigureAwait(false);
+
+                            TryInitializeResponseContentLength();
+
+                            if (_responseContentLength.HasValue && _responseBytesWritten < _responseContentLength.Value)
+                            {
+                                ThrowContentLengthMismatchException();
```

Comment:
Why even throw at this point? Why not just directly invoke ReportApplicationError?